helpers.lua

```diff
@@ -1058,6 +1050,15 @@ end
 -- Set cookie and redirect (needed to properly set cookie)
 function redirect(url)
     logger.debug("Redirecting to "..url)
+    -- For security reason we don't allow to redirect onto unknown domain
+    -- And if `uri_args.r` contains line break, someone is probably trying to
+    -- pass some additional headers
+    local domain = url:match("^https?://([%w%.]*)/?")
+    if string.match(url, "(.*)\n") or not is_in_table(conf["domains"], domain) then
+        logger.debug("Unauthorized redirection to "..url)
+        flash("fail", t("redirection_error_invalid_url"))
+        url = conf.portal_url
+    end
     return ngx.redirect(url)
```

Comment:
It looks like in some cases, our `redirect()` is used to redirect to an uri, not an url. such has here https://github.com/YunoHost/SSOwat/blob/dev/access.lua#L57 and 1 or 2 other places ?

Reply:
@zamentur is this PR still under development ?

Reply:
bumpitybump

Reply:
definitely i will fix that today...

Reply:
should be fixed